server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -91,6 +91,12 @@ public void checkPermission(Permission perm) {
         final Elasticsearch elasticsearch = new Elasticsearch();
         int status = main(args, elasticsearch, Terminal.DEFAULT);
         if (status != ExitCodes.OK) {
+            Terminal.DEFAULT.errorPrintln("ERROR: Elasticsearch did not exit normally - check the logs at " +
```

Comment:
I think it's possible that we can fail *before* these system properties have been set.

Reply:
The problem here is that we can fail before we’ve configured logging, and so those system properties would not be set giving an empty error message to the user. The guarantee that we have here though is that if logging is not yet been configured, then no log messages will have been written (we have a check that ensures we don't touch logging before it's configured, and we fail startup if it is). That means that we should skip this message if the logging properties are not set, because we know from our guarantees that there won't be anything in the logs to look at.

Reply:
@jasontedor I've added guards around both places where I'm printing the log file location.